### PR TITLE
feat: validating data and refactoring code by compartmentalizing logic so controllers can be cleaner

### PR DESCRIPTION
**Pull request by marilia-calado**
It kinda does what is in the commit title

### CODE REVIEW

File: app/controllers/user_controller.py [hunk 1]
```diff
@@ -126,29 +125,13 @@ def delete():
...
-            if key in authorized_keys:
-                setattr(user, key, value)
-            else:
-                invalid_keys.append(key)
-
-        if invalid_keys:
-            raise InvalidKeysError(authorized_keys, invalid_keys)
+        user = get_user_from_token()
+        validate_type(data)
+        validate_and_setattr(data, user)
```
Comment (marcelosj3) [inline app/controllers/user_controller.py]: You could put these two validate functions inside a `validate_keys_values` function (or something like this) and call just once in here, also you could instantiate the `authorized keys` list in here and pass as parameter to this function.

File: app/services/user_services.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from app.exceptions.generic_exc import InvalidKeysError, InvalidTypeError
+from app.models.user_model import UserModel
+
+
+def validate_and_setattr(data: dict, user: UserModel):
+    authorized_keys = ["email", "phone", "name", "password"]
```
Comment (marcelosj3) [inline app/services/user_services.py]: I think it's better to not write this list two times, create it once in the user controller and pass as a parameter to both functions.

File: app/services/user_services.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            setattr(user, key, value)
+        else:
+            invalid_keys.append(key)
+
+    if invalid_keys:
+        raise InvalidKeysError(authorized_keys, invalid_keys)
+
+
+def validate_type(data: dict):
+    authorized_keys = ["email", "phone", "name", "password"]
```
Comment (marcelosj3) [inline app/services/user_services.py]: Same as above.

**Review by marcelosj3 — CHANGES_REQUESTED** [review]

**Review by marcelosj3 — APPROVED** [review]

File: app/controllers/user_controller.py (hunk 1 above)
Comment (marilia-calado) [reply app/controllers/user_controller.py]: Done

**Review by penguinuux — APPROVED** [review]

**Pull request merged**